src/SprykerEco/Zed/Unzer/Business/Checkout/Mapper/UnzerCheckoutPostSaveMapper.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/**
+ * MIT License
+ * For full license information, please view the LICENSE file that was distributed with this source code.
+ */
+
+namespace SprykerEco\Zed\Unzer\Business\Checkout\Mapper;
+
+use ArrayObject;
+use Generated\Shared\Transfer\ItemTransfer;
+use Generated\Shared\Transfer\QuoteTransfer;
+use Generated\Shared\Transfer\UnzerBasketItemTransfer;
+use Generated\Shared\Transfer\UnzerBasketTransfer;
+use Generated\Shared\Transfer\UnzerPaymentResourceTransfer;
+use SprykerEco\Zed\Unzer\UnzerConfig;
+
+class UnzerCheckoutPostSaveMapper implements UnzerCheckoutMapperInterface
+{
+    /**
+     * @var \SprykerEco\Zed\Unzer\UnzerConfig
+     */
+    protected $unzerConfig;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\UnzerConfig $unzerConfig
+     */
+    public function __construct(UnzerConfig $unzerConfig)
+    {
+        $this->unzerConfig = $unzerConfig;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\UnzerBasketTransfer $unzerBasketTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerBasketTransfer
+     */
+    public function mapQuoteTransferToUnzerBasketTransfer(
+        QuoteTransfer $quoteTransfer,
+        UnzerBasketTransfer $unzerBasketTransfer
+    ): UnzerBasketTransfer {
+        return $unzerBasketTransfer
+            ->setAmountTotalGross($quoteTransfer->getTotals()->getGrandTotal() / 100)
+            ->setAmountTotalVat($quoteTransfer->getTotals()->getTaxTotal()->getAmount() / 100)
+            ->setCurrencyCode($quoteTransfer->getCurrency()->getCode())
+            ->setNote('')
+            ->setOrderId($quoteTransfer->getPayment()->getUnzerPayment()->getOrderId())
+            ->setBasketItems($this->mapQuoteItemsCollectionToBasketItemsCollection($quoteTransfer));
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     * @param \Generated\Shared\Transfer\UnzerPaymentResourceTransfer $unzerPaymentResourceTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerPaymentResourceTransfer
+     */
+    public function mapQuoteTransferToUnzerPaymentResourceTransfer(
+        QuoteTransfer $quoteTransfer,
+        UnzerPaymentResourceTransfer $unzerPaymentResourceTransfer
+    ): UnzerPaymentResourceTransfer {
+        return $unzerPaymentResourceTransfer->setType(
+            $this->mapUnzerPaymentType($quoteTransfer->getPayment()->getPaymentSelection())
+        );
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \ArrayObject|\Generated\Shared\Transfer\UnzerBasketItemTransfer[]
+     */
+    protected function mapQuoteItemsCollectionToBasketItemsCollection(QuoteTransfer $quoteTransfer): ArrayObject
+    {
+        $basketItems = new ArrayObject();
```

Comment:
```suggestion
        $basketItemTransfers = new ArrayObject();
```

From @dmytro-sokolovskyi: Better use `$unzerBasketItemTransferCollection` as I mentioned in another comment to align with transfer and method names.